src/extensions/default/analyzers/Microsoft.DotNet.UpgradeAssistant.Extensions.Default.Analyzers/NameMatcher.cs

```diff
@@ -19,10 +19,37 @@ private NameMatcher(string typeName, string? memberName = null)
             _memberName = memberName;
         }
 
+        public static NameMatcher MatchType(string typeName) => new NameMatcher(typeName);
+
         public static NameMatcher MatchPropertyAccess(string typeName, string memberName) => new(typeName, memberName);
 
         public string TypeName => _typeName[_typeName.Length - 1];
 
+        public bool MatchesPartiallyQualifiedType(string partiallyQualifiedTypeName)
```

Comment:
this is invoked from the analyzer for each evaluation. Is there a way we can avoid doing a string.Split('.') on every evaluation?

Reply:
Actually, I've changed my mind. I'll just add the caching now. It's unclear to me whether it will matter, but it's easy enough to add.